app/src/main/kotlin/com/simplemobiletools/calculator/activities/MoneyActivity.kt

```diff
@@ -73,6 +86,38 @@ class MoneyActivity : SimpleActivity(), Calculator , TaxCalculator {
         }
     }
 
+    override fun spawnCurrencyModal() {
+        currencyDialog = AlertDialog.Builder(this)
+        val currencyDialogView = layoutInflater.inflate(R.layout.currency_modal, null)
+        currencyDialog!!.setView(currencyDialogView)
+        currencyDialog!!.setCancelable(true)
+        var custom_dialog = currencyDialog!!.create()
+        custom_dialog.show()
+        var convert_from = "CAD"
+        var convert_to = "CAD"
+
+        custom_dialog.findViewById<Spinner>(R.id.convert_from).onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
```

Comment:
We should look into having one listener for both spinners. Lots of duplicate code here

Reply:
When an item is selected one listener sets the "convert_from" field, and the other sets the "convert_to" field. Seems appropriate two have two listeners because the two spinners are modifying different fields.